pages/api/Schema/reports/designationAccounts/datahandler.ts

```diff
@@ -38,12 +42,69 @@ export interface DesignationAccountsResponse {
   };
 }
 
-export const mapDesignationAccounts = (
+type PreDesignationAccountsGroup = {
+  [organizationId: string]: DesignationAccount[];
+};
+
+const createDesignationAccount = (
+  account: DesignationAccountsResponse,
+): DesignationAccount => ({
+  active: account.attributes.active,
+  balanceUpdatedAt: account.attributes.balance_updated_at,
+  currency: account.attributes.currency,
+  designationNumber: account.attributes.designation_number,
+  id: account.id,
+  name: account.attributes.name,
+  convertedBalance: account.attributes.converted_balance,
+});
+
+export const createDesignationAccountsGroup = (
   data: DesignationAccountsResponse[],
-): DesignationAccount[] =>
-  data.map((account) => ({
-    active: account.attributes.active,
-    currency: account.attributes.currency,
-    name: account.attributes.name,
-    convertedBalance: account.attributes.converted_balance,
+): DesignationAccountsGroup[] => {
+  const preDesignationAccountsGroup = data.reduce(
+    (obj: PreDesignationAccountsGroup, item) => {
+      obj[item.relationships.organization.data.id] =
+        obj[item.relationships.organization.data.id] || [];
+      obj[item.relationships.organization.data.id].push(
+        createDesignationAccount(item),
+      );
+      return obj;
```

Comment:
If you think something like this is any cleaner, feel free to use it. Otherwise what you've got is fine.
```suggestion
      return {
        ...obj,
        [item.relationships.organization.data.id]: [
          ...(obj[item.relationships.organization.data.id] || []),
          createDesignationAccount(item),
        ],
      };
```